rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto

```diff
@@ -156,7 +156,18 @@ message VirtualMachineRunCommandResponse {
 	VirtualMachineRunCommandInstanceView InstanceView = 2;
 }
 
+message VirtualMachineCapacityCheckRequest {
+	repeated VirtualMachine VirtualMachines = 1;
+}
+
+message VirtualMachineCapacityCheckResponse {
+	google.protobuf.BoolValue Result = 1;
+	string Error = 2;
+	map<string, string> Placement = 3;
+}
+
 service VirtualMachineAgent {
+	rpc CheckCapacity(VirtualMachineCapacityCheckRequest) returns (VirtualMachineCapacityCheckResponse) {}
```

Comment:
Should we name this more generic like precheck. I think we can reuse the same to validate other checks like if the gallery image is healthy, storage container is healthy. We can explore on making the status more specific. Thoughts?

Reply:
OK. Will update.